google-cloud-core/src/main/java/com/google/cloud/Binding.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class Binding {
```

Comment:
Yes, and I could add Beta annotation if that's what you're thinking?

Reply:
depends on how confident you are in this. Once we've  published it, it's very hard to change later.

Reply:
Updated to use BetaApi

Reply:
Not sure but can we use Guava's `@Beta` instead? I'd like top deprecate `@BetaApi` one of these days since we really shouldn't have separate annotations for the same thing.

https://github.com/googleapis/api-common-java/issues/101

Reply:
Updated, thank you!

Reply:
Is this considered having guava types on the surface of our library?

Reply:
It has class retention, not runtime retention, so I'd argue no. Also, see

https://stackoverflow.com/questions/3567413/why-doesnt-a-missing-annotation-cause-a-classnotfoundexception-at-runtime